database/tests/audit_test.go

```diff
@@ -0,0 +1,116 @@
+package tests
+
+import (
+	"fmt"
+	"path"
+	"path/filepath"
+	"sync"
+	"testing"
+
+	"github.com/thrasher-corp/gocryptotrader/database"
+	"github.com/thrasher-corp/gocryptotrader/database/drivers"
+	mg "github.com/thrasher-corp/gocryptotrader/database/migration"
+	"github.com/thrasher-corp/gocryptotrader/database/repository/audit"
+	auditPSQL "github.com/thrasher-corp/gocryptotrader/database/repository/audit/postgres"
+	auditSQlite "github.com/thrasher-corp/gocryptotrader/database/repository/audit/sqlite"
+)
+
+func TestAudit(t *testing.T) {
+	testCases := []struct {
+		name   string
+		config database.Config
+		audit  audit.Repository
+		runner func(t *testing.T)
+		output interface{}
+	}{
+		{
+			"SQLite",
+			database.Config{
+				Driver:            "sqlite",
+				ConnectionDetails: drivers.ConnectionDetails{Database: path.Join(tempDir, "./testdb.db")},
```

Comment:
Hopefully the last nit, but for me with windows, I get the following output when running the test here:
> Failed to remove temp db file: remove C:\Users\SCOTTG~1\AppData\Local\Temp\gct-temp080595351\testdb.db: The process cannot access the file because it is being used by another process.


Is this due to the removal of close? >_>

Reply:
![image](https://user-images.githubusercontent.com/1223381/63317102-de8d9280-c354-11e9-9819-a3161851c650.png)

This is me right now.

Will add a Close func that can be configured per subtest